kiwi/config/functions.sh

```diff
@@ -1137,4 +1137,65 @@ function baseQuoteFile {
     mv "${conf}" "${file}"
 }
 
+#======================================
+# baseVagrantSetup
+#--------------------------------------
+function baseVagrantSetup {
+    # insert the default insecure ssh key from here:
+    # https://github.com/hashicorp/vagrant/blob/master/keys/vagrant.pub
+    mkdir -p /home/vagrant/.ssh/
+    echo "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key" > /home/vagrant/.ssh/authorized_keys
+    chmod 0600 /home/vagrant/.ssh/authorized_keys
+    chown -R vagrant:vagrant /home/vagrant/
+
+    # recommended ssh settings for vagrant boxes
+    echo "UseDNS no" >> /etc/ssh/sshd_config
+    echo "GSSAPIAuthentication no" >> /etc/ssh/sshd_config
+
+    # vagrant assumes that it can sudo without a password
+    # => add the vagrant user to the sudoers list
+    SUDOERS_LINE="vagrant ALL=(ALL) NOPASSWD: ALL"
+    if [ -d /etc/sudoers.d ]; then
+        echo "$SUDOERS_LINE" >| /etc/sudoers.d/vagrant
+        visudo -cf /etc/sudoers.d/vagrant
+        chmod 0440 /etc/sudoers.d/vagrant
+    else
+        echo "$SUDOERS_LINE" >> /etc/sudoers
+        visudo -cf /etc/sudoers
+    fi
+
+    # the default shared folder
+    mkdir -p /vagrant
+    chown -R vagrant:vagrant /vagrant
+
+    # SSH service
+    baseInsertService sshd
+}
+
+#======================================
+# suseVagrantSetup
+#--------------------------------------
+function suseVagrantSetup {
+    baseVagrantSetup
+
+    # start vboxsf service only if the guest tools are present
+    if Rpm -q virtualbox-guest-tools 2> /dev/null; then
+        echo vboxsf > /etc/modules-load.d/vboxsf.conf
+    fi
+
+    # drop any network udev rules for libvirt, so that the networks are called
+    # ethX
+    # this is not required for Virtualbox as it handles networking differently
+    # and doesn't need this hack
+    if [ "${kiwi_profiles}" != "virtualbox" ]; then
```

Comment:
I don't think this should behave differently depending on profile names. What about a `suseSetupDHCP` function which the script can call?

Reply:
That step is only required for vagrant-libvirt, as that uses dnsmasq for assigning IPs. Virtualbox does that differently and the modifications are not necessary.

Reply:
The generic method should not depend on a profile name. The setup of a profile is an optional setting and it can be used as part of script code that belongs to the rest of an image description which has the profile configured. However for a generic method considered a helper for people building vagrant images you can't expect the profile "virtualbox" to exist.

Furthermore the code would kick in if a profile is named "virtualbox" but it's not guaranteed that this has something to do with your context. Agreed this is a constructed example and very unlikely to happen but any code provided as generic helpers should not be image description specific

Reply:
> What about a suseSetupDHCP function which the script can call?

sounds like a good approach to me too